src/test/java/net/sf/jabref/importer/fileformat/SilverPlatterImporterTest.java

```diff
@@ -0,0 +1,85 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.bibtex.BibtexEntryAssert;
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.BibtexEntryTypes;
+
+public class SilverPlatterImporterTest {
+
+    private SilverPlatterImporter testImporter;
+
+
+    @Before
+    public void setUp() throws Exception {
+        Globals.prefs = JabRefPreferences.getInstance();
+        testImporter = new SilverPlatterImporter();
+    }
+
+    @Test
+    public final void testIsRecognizedFormat() throws Exception {
+        try (InputStream stream = SilverPlatterImporterTest.class
+                .getResourceAsStream("SilverPlatterImporterTest1.txt")) {
+            Assert.assertTrue(testImporter.isRecognizedFormat(stream));
+        }
+    }
+
+    @Test
+    public final void testIsNotRecognizedFormat() throws Exception {
+        List<String> notAccepted = Arrays.asList("emptyFile.xml", "IsiImporterTest1.isi",
+                "InspecSilverPlatterImporterTest.txt", "oai2.xml", "RisImporterTest1.ris");
+        for (String s : notAccepted) {
+            try (InputStream stream = SilverPlatterImporterTest.class.getResourceAsStream(s)) {
+                Assert.assertFalse(testImporter.isRecognizedFormat(stream));
+            }
+        }
+    }
+
+    @Test
+    public final void testImportEntries1() throws Exception {
+
+        try (InputStream in = SilverPlatterImporter.class.getResourceAsStream("SilverPlatterImporterTest1.txt");
+                InputStream bibIn = SilverPlatterImporter.class.getResourceAsStream("SilverPlatterImporterTest1.bib")) {
+            List<BibEntry> entries = testImporter.importEntries(in, new OutputPrinterToNull());
+            Assert.assertEquals(1, entries.size());
+            BibtexEntryAssert.assertEquals(bibIn, entries.get(0));
+        }
+
+    }
+
+    @Test
+    public final void testImportEntries2() throws Exception {
+        try (InputStream in = SilverPlatterImporter.class.getResourceAsStream("SilverPlatterImporterTest2.txt")) {
+            List<BibEntry> entries = testImporter.importEntries(in, new OutputPrinterToNull());
+            Assert.assertEquals(1, entries.size());
+            Assert.assertEquals(BibtexEntryTypes.ARTICLE.getName().toLowerCase(), entries.get(0).getType());
+            Assert.assertEquals("New York publishment", entries.get(0).getField("publisher"));
+            Assert.assertEquals("3 Jun", entries.get(0).getField("year"));
+            Assert.assertEquals("many-much-more-i don't know", entries.get(0).getField("keywords"));
+            Assert.assertEquals("Gymnasium Unterrieden", entries.get(0).getField("school"));
+            Assert.assertEquals("Supek-S and Aine-CJ", entries.get(0).getField("editor"));
+            Assert.assertEquals("IEEE Trans Biomed Eng", entries.get(0).getField("journal"));
```

Comment:
Better use the BibTexAssert.assertEquals(Collections.singletonList(shouldBeEntry), entries). See the other PRs